pkg/querier/block_store.go

```diff
@@ -76,8 +103,34 @@ func (u *UserStore) InitialSync(ctx context.Context) error {
 	return nil
 }
 
-// SyncStores iterates over the storage bucket creating user bucket stores
-func (u *UserStore) SyncStores(ctx context.Context) error {
+// syncStoresLoop periodically calls syncStores() to synchronize the blocks for all tenants.
+func (u *UserStore) syncStoresLoop() {
```

Comment:
This was existent in old code, but since `runutil.Repeat` runs f right after being called, we effectively sync twice immediately. I think it might make sense to put the initial sync in this function. Then have a `time.After` where it waits to start the repeat function, so there's a little room between initial sync and the repeating sync. Thoughts?

Reply:
I think you're right. Will work on it.